#11 Crear un template para los issues

LPRegen opened the pull request:
<!--
Antes de realizar el Pull Request, por favor asegúrate de haber realizado lo siguiente:
  - 📖 Haber leído el Código de Conduta https://github.com/Juguetear/juguetear-web#c%C3%B3digo-de-conducta
  - 📖 Haber leído Cómo contribuir https://github.com/Juguetear/juguetear-web/blob/development/CONTRIBUTE.md

 ⛑️ En caso de que necesites ayuda o tengas alguna duda recordá que también podés hacerlo a través de nuestro servidor de Discord https://discord.com/invite/frontendcafe y luego busques el canal #juguetear. El código de conducta de este proyecto es extensible también a tu participación en el server de FrontendCafé en Discord.

Más información respecto a la sintaxis para archivos md, https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
 -->

# Issue:

<!--
Dependiendo el caso, completar según corresponda con el link al issue resuelto/relacionado y eliminar el que no.
-->

- Resuelve issue: [Crear un template para los issues #11](https://github.com/Juguetear/juguetear-web/issues/11)

## Qué tipo de PR es esta? (marcar las que correspondan)

<!--
Reemplazar el espacio dentro de los corchetes por una X para seleccionar un elemento de la lista.
-->

- [ ] Refactor.
<!-- Modificación de código existente -->

- [x] Nueva funcionalidad.
<!-- Introducción de nueva funcionalidad / Feature -->

- [ ] Corrección de error/es.
<!-- Corrección de errores / Bug fix -->

- [ ] Optimización.
<!-- Cambios relacionados a performance -->

- [x] Actualización/extensión de documentación.
<!-- Cambios en la documentación / Documentation related. -->

## Descripción

<!--
Explicar el porqué de las decisiones que se tomaron y el objetivo de las mismas.
Por ejemplo, si se implementa una nueva librería incluír los fundamentos de porqué fué elegida sobre otras (peso de la misma, funcionalidad, etc.), si se realiza un refactor de código ya existente indicar las ventajas potenciales de la modificación realizada.
-->

Se realiza modificación de CONTRIBUTE.md. Se explica al usuario las diferentes maneras de realizar contribuciones al proyecto y se agrega información respecto a la licencia bajo la cual se encuentra el mismo.

Se agregan templates para la creación de nuevos issues y solicitudes de features.
Cuando el usuario se dirige a crear un nuevo issue tiene tres opciones, crear un feature, reportar un bug o dirigirse a FEC.

**Bugs/issues template:**
- Checkbox Código de Conducta, Licencia, comprobar que no haya sido reportado. (requerido)
- Detalles de contacto (opcional)
- Bug encontrado (requerido)
- Como replicarlo (requerido)
- Comportamiento esperado (requerido)
- Comportamiento actual (requerido)
- Navegador en el que sucede (requerido)
- Otro navegador que estés utilizando (opcional)
- Log (opcional) 

**Feature template:**
- Checkbox Código de Conducta, Licencia, comprobar que no haya sido reportado. (requerido)
- Detalles de contacto (opcional)
- Feature propuesta (requerido)
- Requerimientos de la mejora (requerido)
- Implementación (requerido)

**Por completar:**
- [ ] Agregar link directo al canal "#juguetear".
- [x] Actualizar link directo para reportar un bug.
- [x] Actualizar link directo para proponer una feature.

**Completado:**
- [X] Crear "triage" label, utilizado para features que deben ser discutidas entre administradores/staff. 
- [X] Crear template para features.
- [X] Crear template para issues.

### Links

No aplica.

<!--
En caso de que creas necesario añadir links a documentación, recursos, etc que estén relacionados.
Si no es necesario colocar "No aplica".
-->

### Pasos necesarios para testear/visualizar los cambios implementados

El archivo se encuentra en [https://github.com/Juguetear/juguetear-web/blob/development/CONTRIBUTE.md](https://github.com/Juguetear/juguetear-web/blob/development/CONTRIBUTE.md)

El resultado de los templates se puede visualizar una vez que sean mergeados a development y se intente crear un nuevo issue/feature.

<!--
Incluir los requerimientos para testear/visualizar los cambios. Por ejemplo, si implementa una nueva librería incuir los comandos necesarios para instalar la misma.
Si no es necesario colocar "No aplica".
-->

## Archivos modificados/creados

Modificados:

- CONTRIBUTE.md
- .github/ISSUE_TEMPLATE/
> BUG_FORM.yml
> FEATURE_FORM.yml
> config.yml

<!--
Utilizar un bullet (-) por archivo modificado/creado.
-->

## Capturas de pantalla/videos

<!--
Seleccionar según corresponda:
-->

- [ ] Nuevo componente.
<!-- Incluir capturas de pantalla en diferentes dispositivos -->

- [ ] Modificación a elemento de User Interface.
<!-- Incluir captura con estado previo y posterior a la modificación indicando cuál corresponde a cada una. -->

- [x] No aplica.
<!-- En el caso de que sea un refactor a la lógica por ejemplo. -->

LPRegen committed:
docs: refactor CONTRIBUTE file

TonyMckes commented on CONTRIBUTE.md:
```diff
@@ -1,63 +1,81 @@
+En este archivo, encontrarás información sobre las diferentes maneras de realizar contribuciones a este proyecto. El código se encuentra almacenado en GitHub, utilizamos el mismo para llevar el seguimiento de reportes de errores (bugs) y proponer mejoras (features).
 
-### Reportar un _bug_
+Si en algún momento necesitas hacer una consulta respecto al proyecto o el código del mismo, podés ingresar a nuestro [servidor en Discord](https://discord.com/invite/frontendcafe) de FrontendCafé, buscar el canal **#juguetear** y realizar todas las consultas que sean necesarias.
```
@LPRegen  se me ocurrio que si el canal #juguetear sera permanente, quizas se pueda agregar un perma-url, o quizas @nykka pueda crear una invitacion directa a ese canal (no se como es la cosa), asi las personas pueden llegar directamente.

LPRegen replied:
Ah si eso estaría genial, no sabía que se podía hacer eso!

nykka replied:
Si, lo vamos a hacer cuando el canal esté público. Asi que por ahora lo dejamos así y cuando esté listo, lo actualizamos

LPRegen replied:
Genial

LPRegen committed:
feat: create issue template

LPRegen committed:
feat: craete feature template

LPRegen committed:
feat: add config for templates

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+    id: terms
+    attributes:
+      label: Código de conducta.
+      description: Al completar esta solicitud, aceptas el [Código de Conducta](https://github.com/Juguetear/juguetear-web#c%C3%B3digo-de-conducta)
```
"aceptás"

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+    id: terms
+    attributes:
+      label: Licencia.
+      description: Al completar esta solicitud, aceptas la [licencia del repositorio](https://github.com/Juguetear/juguetear-web/blob/development/LICENSE)
```
"aceptás"

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+      label: He comprobado que no se haya solicitado la feature.
+      description: Has comprobado que la feature solicitada por vos
+      options:
+        - label: Estoy de acuerdo con la licencia de este proyecto.
```
Aca creo que te olvidaste de actualizar el label

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+    id: terms
+    attributes:
+      label: He comprobado que no se haya solicitado la feature.
+      description: Has comprobado que la feature solicitada por vos
```
Acá falta texto, no?

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+    id: contact
+    attributes:
+      label: Detalles de contacto
+      description: ¿Cómo podemos contactarte en caso de ser necesario? No es mandatorio.
```
En vez de "no es mandatorio", pondría "este campo es opcional"

nykka commented on .github/ISSUE_TEMPLATE/BUG_FORM.yml:
```diff
@@ -0,0 +1,98 @@
+  - type: textarea
+    id: log
+    attributes:
+      label: Por favor completa con logs relevantes respecto al bug.
```
"completá"

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+    id: terms
+    attributes:
+      label: Código de conducta.
+      description: Al completar esta solicitud, aceptas el [Código de Conducta](https://github.com/Juguetear/juguetear-web#c%C3%B3digo-de-conducta)
```
"aceptás"

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+    id: terms
+    attributes:
+      label: Licencia.
+      description: Al completar esta solicitud, aceptas la [licencia del repositorio](https://github.com/Juguetear/juguetear-web/blob/development/LICENSE)
```
"aceptás"

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+    id: terms
+    attributes:
+      label: He comprobado que no se haya solicitado la feature.
+      description: Has comprobado que la feature solicitada por vos
```
Lo mismo que en el anterior, creo que falta texto

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+      label: He comprobado que no se haya solicitado la feature.
+      description: Has comprobado que la feature solicitada por vos
+      options:
+        - label: Estoy de acuerdo con la licencia de este proyecto.
```
También lo mismo que la anterior

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+    id: contact
+    attributes:
+      label: Detalles de contacto
+      description: ¿Cómo podemos contactarte en caso de ser necesario? No es mandatorio.
```
Lo mismo acá

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+    id: feat-description
+    attributes:
+      label: ¿Qué te gustaría agregar a este proyecto?
+      description: Por favor cuéntanos qué es lo que consideras que falta?
```
"contanos"

nykka commented on .github/ISSUE_TEMPLATE/FEATURE_FORM.yml:
```diff
@@ -0,0 +1,89 @@
+      label: ¿Qué te gustaría agregar a este proyecto?
+      description: Por favor cuéntanos qué es lo que consideras que falta?
+      placeholder: Estamos ansiosos por saberlo :)
+      value: 'Contános lo que te gustaría!'
```
Creo que contanos va sin tilde

nykka reviewed (changes requested):
Lo veo genial, te dejé unos comentarios de tipo ortografico/gramatica/estilo nomás

LPRegen committed:
refactor: apply requested changes

- apply requested changes
- solve error of checkboxes' ids

LPRegen committed:
docs: refactor CONTRIBUTE and issue templates

- Add templates description to CONTRIBUTE
- refactor remove textarea from FEATURE_FORM
- refactor checkbox for licenceses FEATURE_FORM and BUG_FORM

nykka reviewed (approved):
LGTM!
